src/dune_engine/build.ml

```diff
@@ -10,6 +10,7 @@ type 'a t =
   | Paths_for_rule : Path.Set.t -> unit t
   | Paths_glob : File_selector.t -> Path.Set.t t
   | If_file_exists : Path.t * 'a t * 'a t -> 'a t
+  | Filter_existing_files : ('a * Path.Set.t) t -> ('a * Path.Set.t) t
```

Comment:
It's not immediately obvious why `Filter_existing_files` can't be built on top of `If_file_exists`, so a short comment would be useful for future readers.

Reply:
Yes, that's a very clear explanation. It's exactly how I convinced myself after being puzzled for a couple of minutes.